### PR TITLE
feat(Forms): add `reduceToVisibleFields` to the `useData` hook and Form.Handler `onSubmit`

### PR DESCRIPTION
**Pull request by tujoworker**
This PR adds a new method that lets a dev remove all data which is not given by a mounted/visible field.

Quick example:

```tsx
<Form.Handler
  onSubmit={(data, { reduceToVisibleFields }) => {
    const myData = reduceToVisibleFields(data)
  }}
>
...
</Form.Handler>
```

### CODE REVIEW

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Form/getData/info.mdx [hunk 1]
```diff
@@ -35,6 +37,28 @@ Related helpers:
...
+}
+
+// Later, when there is data available
+const { data, reduceToVisibleFields } = Form.getData('unique-id')
+const visibleData = reduceToVisibleFields(data)
+```
+
 ## Filter data
 
 You can use the `filterData` method to filter your data.
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Form/getData/info.mdx:64]:
```suggestion
You can use the `filterData` function to filter your data.
```

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Form/Handler/Examples.tsx [hunk 2]
```diff
@@ -297,6 +297,48 @@ export const Locale = () => {
...
 
+export const VisibleData = () => {
+  return (
+    <ComponentBox>
+      <Form.Handler
+        onSubmit={(data, { reduceToVisibleFields }) => {
+          const myData = reduceToVisibleFields(data, {
+            removePaths: ['/isVisible'],
+          })
+          console.log('Result of reduceToVisibleFields:\n', myData)
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Form/Handler/Examples.tsx:308]:
```suggestion
          console.log('Result of reduceToVisibleFields: ', myData)
```

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Form/useData/info.mdx [hunk 3]
```diff
@@ -131,9 +149,43 @@ function MyForm() {
 }
 ```
 
+## Visible data
+
+You can use the `reduceToVisibleFields` method to get only the data of visible (mounted) fields. Check out the example in the demo section.
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Form/useData/info.mdx:154]:
```suggestion
You can use the `reduceToVisibleFields` method to get only the data of visible (mounted) fields. Check out the [example](/uilib/extensions/forms/Form/Handler/demos/#visible-data) in the demo section.
```

**Comment by langz** [comment]
When entering the example, https://eufemia-git-feat-get-visible-fields-data-eufemia.vercel.app/uilib/extensions/forms/Form/Handler/demos/#visible-data, there's a lot that's happening(flickering, changing of values between "ja" and "nei", and then displaying radio buttons with animation then display the `<Value.Selection />`):

https://github.com/user-attachments/assets/11a82610-8da5-4e9c-9105-5fe36b0c0bac

**Comment by langz** [comment]
I made a [CSB based on a commit in this branch/pr](https://codesandbox.io/p/sandbox/ecstatic-napier-yz39jv?workspaceId=9eb653cb-4406-4b8f-bc96-714927cc7fcd), and I see some TS error(`Property 'reduceToVisibleFields' does not exist on type 'OnSubmitParams'.typescript(2339)
`) and I did not get the expected result when doing `reduceToVisibleFields(data)`.

**Comment by tujoworker** [comment]
> I would expect to get some data in the following scenario, but I get an empty object

Nice! Thank you. This is due to React StrictMode. I need to look into this.

**Comment by langz** [comment]
> > I would expect to get some data in the following scenario, but I get an empty object
> 
> Nice! Thank you. This is due to React StrictMode. I need to look into this.

Thanks for improving, I've now made a new [CSB](https://codesandbox.io/p/sandbox/wizardly-burnell-ljx6qn?workspaceId=9eb653cb-4406-4b8f-bc96-714927cc7fcd) based on the [latest commit](https://github.com/dnbexperience/eufemia/pull/3957/commits/14dae9fd819be076a0e9977b5cf6518c56a9dac4), and it works great ❤️

**Comment by langz** [comment]
> When entering the example, https://eufemia-git-feat-get-visible-fields-data-eufemia.vercel.app/uilib/extensions/forms/Form/Handler/demos/#visible-data, or when refreshing the browser, there's a lot that's happening(flickering, changing of values between "ja" and "nei", and then displaying radio buttons with animation then display the `<Value.Selection />`):
> 
>  Screen.Recording.2024-09-18.at.13.28.07.mov

As long as what I experience, is most likely only possible to experience in the portal, then we shouldn't need to look into improving this. However, if we think this may happen in real application code, then we should perhaps look into it 🙇

**Comment by langz** [comment]
I've added a commit https://github.com/dnbexperience/eufemia/pull/3957/commits/a69769d81fd55fa480b7143e58e8394076c57f23, to update the tests based on the changes in your latest commit https://github.com/dnbexperience/eufemia/pull/3957/commits/14dae9fd819be076a0e9977b5cf6518c56a9dac4

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Form/useData/info.mdx [hunk 4]
```diff
@@ -131,9 +149,43 @@ function MyForm() {
 }
 ```
 
+## Visible data
+
+You can use the `reduceToVisibleFields` method to get only the data of visible (mounted) fields. Check out the [example](/uilib/extensions/forms/Form/Handler/demos/#visible-data) in the demo section.
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Form/useData/info.mdx:154]:
```suggestion
You can use the `reduceToVisibleFields` function to get only the data of visible (mounted) fields. Check out the [example](/uilib/extensions/forms/Form/Handler/demos/#visible-data) in the demo section.
```

**Pull request merged**